.github/workflows/check-odo.yml

```diff
@@ -5,41 +5,131 @@ on:
     - cron: "0 8 * * *"
   workflow_dispatch:
 jobs:
-  check-odo-repo:
+  should-check-odo:
     runs-on: ubuntu-latest
-    env:
-        TOOL_REPO: redhat-developer/odo
-        GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+    outputs:
+      repo-cache-hit: ${{ steps.cache-last-commit.outputs.cache-hit }}
+      repo-last-commit-id: ${{ steps.repo-last-commit-info.outputs.repo-last-commit-id }}
+      repo-last-commit-date: ${{ steps.repo-last-commit-info.outputs.repo-last-commit-date }}
+      repo-last-commit-build-number: ${{ steps.repo-last-commit-info.outputs.repo-last-commit-build-number }}
+      repo-last-release-tag: ${{ steps.repo-last-commit-info.outputs.repo-last-release-tag }}
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          repository: 'redhat-developer/odo'
+          fetch-depth: 0
+          fetch-tags: true
+          path: redhat-developer-odo-repository
+      - name: Get Last Commit Info
+        id: repo-last-commit-info
+        run: |
+          pushd redhat-developer-odo-repository
+
+          # Last Commit ID
+          lastCommitId="$(git rev-parse HEAD 2>/dev/null)"
+          echo "repo-last-commit-id=$lastCommitId" >> "$GITHUB_OUTPUT"
+
+          # Last Commit Committer Date/Time
+          echo "repo-last-commit-date=$(git log -1 --oneline --pretty=format:%cd --date=format:%Y%m%d%H%M%S ${lastCommitId} 2>/dev/null)" >> "$GITHUB_OUTPUT"
+
+          # Last Release Tag
+          lastReleaseTag="$(git describe --tags --abbrev=0 2>/dev/null)"
+          echo "repo-last-release-tag=$lastReleaseTag" >> "$GITHUB_OUTPUT"
+
+          # Number of commits between the latest release and the Last Commit ID
+          echo "repo-last-commit-build-number=$(git log --oneline ${lastReleaseTag}..${lastCommitId} 2>/dev/null | wc -l)" >> "$GITHUB_OUTPUT"
```

Comment:
Is this really necessary ? I would get rid of it.

Reply:
This is used to generate nightly build version: "3.15.**1**-**_116_**-nightly"  - here `116` is the value of `repo-last-commit-build-number` 

For each nightly build we need to create an unique version  which will `semver` compatible and, when sorting, is to be placed between the last release version and the [possible] next release version, so I've used this build number as a qualifier. 

As an alternative we can use the nightly build last commit creation date, but it's worse as some commits merged on top of the main branch may be created earlier that the previously build nightly - then in theory this might make a problem. So, I'd prefer keep this `repo-last-commit-build-number` value "aka build number qualifier" here

Reply:
Outdated. NO more "nightly build number" nor "commit date" is used